packages/lib-core/src/store/PluginLoader.ts

```diff
@@ -304,10 +343,146 @@ export class PluginLoader {
     });
   }
 
+  /**
+   * Append entry module to DOM
+   */
+  injectPluginScript(
+    baseURL: string,
+    pluginLoadData: PluginLoadData,
+    // onLoad type does not have stand alone definition
+    onLoad?: (ev?: Event) => void,
+    onError?: OnErrorEventHandler,
+    getDocument: () => typeof document = _.constant(document),
+  ) {
+    const pluginName = pluginLoadData.manifest.name;
+
+    this.plugins.set(pluginName, pluginLoadData);
+    const initPromise = new Promise<void>((resolve, reject) => {
+      const script = getDocument().createElement('script');
+
+      script.src = resolveURL(baseURL, pluginLoadData.manifest.entryScript);
+      script.async = true;
+
+      script.onload = async (event) => {
+        if (onLoad) {
+          onLoad(event);
+        }
+        if (pluginLoadData.manifest.libraryType === 'var') {
+          await this.initializeVarModule(pluginName);
+        }
+        return resolve();
```

Comment:
Is this a short-hand for the following?
```ts
resolve();
return;
```

Both `resolve` and `reject` functions return `void`.

Reply:
Yeah, it's just a short hand.